### ISSUE TITLE
Collision Layers

### ISSUE THREAD

**Issue by DindinYT37** [studio-minus/ppg-bugs-and-requests#1684]
**Feature description**
Collision Layers basically mean, only things on the same layer collide. All objects would default to be on Layer 1 though you can set (either by right-clicking and setting it or setting the spawn-layer setting to a different one) the collision layer to a different one.

**Feature purpose**
I don't know - would just be cool I guess. Though this would allow for background and foreground layers of some sort.

**Additional context**
Note that things like explosions should act on every layer, while things like bullets should only act on the layer the gun is on.

**Comment by PossiblyRemember** [comment]
it'd be cool if you had multiple people on different layers fighting on many different layers, I've thought of this for a long time, I really hope this gets added

**Comment by Verycoolgarfield** [comment]
This needs to be in the game

**Comment by DindinYT37** [comment]
> it'd be cool if you had multiple people on different layers fighting on many different layers, I've thought of this for a long time, I really hope this gets added

Also one of the things I've thought of, especially imagine this with the upcoming multiplayer mod. Though to be honest I'm unsure how selecting objects would go, I think it'd probably be a setting that you can modify so you can either interact with objects of all layers or only with the ones on the layer you've currently selected(?).

**Comment by PossiblyRemember** [comment]
> > it'd be cool if you had multiple people on different layers fighting on many different layers, I've thought of this for a long time, I really hope this gets added
> 
> Also one of the things I've thought of, especially imagine this with the upcoming multiplayer mod. Though to be honest I'm unsure how selecting objects would go, I think it'd probably be a setting that you can modify so you can either interact with objects of all layers or only with the ones on the layer you've currently selected(?).

I was thinking only being able to select things on the layer you're on.

**Comment by DindinYT37** [comment]
> I was thinking only being able to select things on the layer you're on.

I thought so too but it'd also make sense if you could toggle being able to select objects from any layer.

**Comment by PossiblyRemember** [comment]
> > I was thinking only being able to select things on the layer you're on.
> 
> I thought so too but it'd also make sense if you could toggle being able to select objects from any layer.

yeah, that'd make sense

**Comment by MrHushPPG** [comment]
> > it'd be cool if you had multiple people on different layers fighting on many different layers, I've thought of this for a long time, I really hope this gets added
> 
> Also one of the things I've thought of, especially imagine this with the upcoming multiplayer mod. Though to be honest I'm unsure how selecting objects would go, I think it'd probably be a setting that you can modify so you can either interact with objects of all layers or only with the ones on the layer you've currently selected(?).

What do you mean by upcoming multiplayer mod? You kind of said that multiplayer shouldn't come out.

**Comment by DindinYT37** [comment]
> > > it'd be cool if you had multiple people on different layers fighting on many different layers, I've thought of this for a long time, I really hope this gets added
> > 
> > 
> > Also one of the things I've thought of, especially imagine this with the upcoming multiplayer mod. Though to be honest I'm unsure how selecting objects would go, I think it'd probably be a setting that you can modify so you can either interact with objects of all layers or only with the ones on the layer you've currently selected(?).
> 
> What do you mean by the upcoming multiplayer mod? You kind of said that multiplayer should not be created.

Yeah but people are still making it, and I'm not intending to stop them in any way. What I meant was that if a multiplayer mod was created and used in that scenario, it'd probably look pretty epic.

**Comment by MrHushPPG** [comment]
> Yeah but people are still making it, and I'm not intending to stop them in any way. What I meant was that if a multiplayer mod was created and used in that scenario, it'd probably look pretty epic.
> 
> Edit: Also I'm pretty sure I just said zooi doesn't want multiplayer, not that I don't want it.

ok, thanks for reply

**Comment by SirDoggoTheBoyo** [comment]
this would be absoloutly amazing in game, would be super useful for contraptions

**Comment by DindinYT37** [comment]
> this would be absoloutly amazing in game, would be super useful for contraptions, maybe as the collision layers get further back or something the sprites get darker to emulate depth?

That'd probably either be another slider or a setting (checkbox) for the layer.
